### PR TITLE
add memo and tx type indexing 

### PR DESCRIPTION
**Pull request by dimiandre**
add memo and tx type indexing for easy queries

### CODE REVIEW

**Comment by rllola** [comment]
I need to fix the tests before accepting your PR. Just to be sure we are not breaking more things.

**Comment by rllola** [comment]
The tests are fixed. You can rebase and I will run them on your PR.

File: src/database.rs [hunk 1]
```diff
@@ -599,7 +601,9 @@ impl Database {
         for t in txs.iter() {
             let tx = Tx::try_from(t.as_slice()).map_err(|_| Error::InvalidTxData)?;
 
-            let mut code = Default::default();
+            let mut code: [u8; 32] = Default::default();
+            let mut code_type: String = "wrapper".to_string();
```
Comment (rllola) [inline src/database.rs:605]: I think this is redundant with `tx_type` which already exist in the table. You want to indicate if the tx is decrypted or wrapper... etc, right ?

Comment (dimiandre) [reply src/database.rs:605]: I'd like to indicate what that decrypted tx actually contains

tx_transfer
tx_ibc
etc

Comment (rllola) [reply src/database.rs:605]: Ah ok. We use the `code` to indicate the tx_type but indeed you need to know it. 
So just change the default value `wrapper` with `unknown`. Wrapper is another type (decrypted, wrapper, protocol,...) and it is different from what you want (aka tx_transfer, tx_ibc, ...). Just so we don't get mixed up.

**Comment by rllola** [comment]
Hi @dimiandre 

If you want me to merge this, could you replace "wrapper" with "unknown" ? I think others will find it useful.

**Review by rllola — APPROVED** [review]

**Comment by rllola** [comment]
Hey @dimiandre 

I am sorry. There is some cnflicts do you think you can solve them ? I can't merge it

**Pull request closed without merging**